benchling-examples-python migration

james-leinas opened the pull request:
Consolidation efforts to include custom-code-in-automation-designer code snippets for public availability.

Changes included:

- Updates to README.md (include references to custom-code-in-automation-designer)
- Add plot-chromatogram example snippet and related docs
- Include requirements.txt dependencies
- Add custom-code-in-automation-designer template.py

james-leinas committed:
Add components for custom-code-in-automation-designer

james-leinas commented:
@damola-benchling -- created PR for migration efforts!

Psterne commented on examples/custom-code-in-automation-designer/snippets/plot-chromatogram/HPLC_Chromatogram_Plot_(Absorbance, pH, Temperature).py:
```diff
@@ -0,0 +1,101 @@
+Supported packages:
+
+allotropy
+benchling-sdk
```
We should probably remove this for now

Psterne commented on examples/custom-code-in-automation-designer/snippets/plot-chromatogram/HPLC_Chromatogram_Plot_(Absorbance, pH, Temperature).py:
```diff
@@ -0,0 +1,101 @@
+
+"""
+Supported packages:
```
We'll have to figure out how to keep this list somewhat in sync with what we have in our manifest. Or add a comment that the versions and packages they get in Benchling might slightly vary from what we have here? Ideally they shouldn't copy paste this into the editor?

Psterne commented on examples/custom-code-AD/snippets/template/template.py:
```diff
@@ -0,0 +1,26 @@
+"""
+Supported packages:
```
We'll have to figure out how to keep this list somewhat in sync with what we have in our manifest. Or add a comment that the versions and packages they get in Benchling might slightly vary from what we have here? Ideally they shouldn't copy paste this into the editor?

damola-benchling commented on README.md:
```diff
@@ -1,6 +1,6 @@
-# Benchling App Python Examples
+# Benchling Python Examples
 
-The `examples/` directory contains Benchling App samples written by Benchling.
```
I think it will be good to structure the README a little more:

I'm thinking we should have H1 for each category, and H2 headers (and short description) for each example in the category (and we can even use [collapsible summary](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections))

# Benchling Apps:
...
## chem-sync-local-flask
Pre-existing description...

# Custom Code in Automation Designer
Contains code snippets, include link to README for snippets

## plot-chromatogram
short description...

damola-benchling commented on README.md:
```diff
@@ -19,4 +19,23 @@ in a local development environment running [Flask](https://flask.palletsprojects
+* Creating [molecule custom entities](https://benchling.com/api/reference#/Molecules/createMolecule)
+
+## Custom Code in Automation Designer
+This project contains example snippets of code that can be utilized in a Custom Code step of the Benchling Automation Designer with Benchling Analysis.
```
Yea I think an overview statement should be sufficient here, and then we can just link to the README in `examples/sippets/README.md` so the rest of the content here will just be in there (rather than duplicated)

damola-benchling reviewed (commented):
I have some minor feedback about the structuring of the READMEs and left comments there.

Also. 
It seems the new structure will look like this:
```sh
examples/
├─ custom-code-in-automation-designer/
│  ├─ snippets/
├─ chem-sync-local-flask/

```

Curious what you think about:
- moving `examples/chem-sync-local-flask` to `examples/benchling-apps/chem-sync-local-flask` now? (I think this was what we said we wanted anyway)
- reducing the length of the name of `custom-code-in-automation-designer`? This is maybe a pet peeve, but long directory names can get gnarly when working in the terminal 😅

james-leinas committed:
updates from comments

james-leinas committed:
fix test

james-leinas commented:
@Psterne @damola-benchling -- Updated the PR with your changes. 

@Psterne -- pulled the current requirements.txt packages from custom code manifest and removed Benchling-SDK references.

@damola-benchling -- 

- updated the structure we discussed to: 

<img width="221" height="190" alt="image" src="https://github.com/user-attachments/assets/b7c064e6-a8a2-434d-9c70-074eb66be935" />

- Renamed `custom-code-in-automation-designer` folder to `custom-code-AD`

- Migrated the README.md content from /custom-code-AD folder to main README.md and added link to example snippets specific README.md

LMK any further feedback!

Psterne commented:
Looks good to me! Looks like only @damola-benchling can approve the PR though.

damola-benchling reviewed (approved):
TY! 
We can continue to iterate on this structure if needed as well 🙏

james-leinas commented:
@damola-benchling -- I think you'll need to squash/merge for me. I don't have those permissions for the repo.

damola-benchling commented:
It's merged @james-leinas sorry I forgot permissions are needed to merge.
